packages/cli/init/src/utils/docsUrlUtils.ts

```diff
@@ -0,0 +1,70 @@
+import axios from "axios";
+import { load } from "cheerio";
+
+export type DocsUrl = ReadMeDocsUrl;
+
+export interface DocsUrlInterface {
+    getAllOpenApiUrls(): Promise<string[]>;
+    getBaseUrl(): string;
+    isUrlValid(): boolean;
+}
+
+class DocsUrlBase implements DocsUrlInterface {
+    url: string;
+
+    constructor(url: string) {
+        this.url = url;
+    }
+
+    async getAllOpenApiUrls(): Promise<string[]> {
+        return [];
+    }
+
+    getBaseUrl(): string {
+        const url = new URL(this.url);
+        return url.origin;
+    }
+
+    isUrlValid(): boolean {
+        if (!this.url || this.url === "") {
+            return false;
+        }
+
+        try {
+            new URL(this.url);
+            return true;
+        } catch (err) {
+            return false;
+        }
+    }
+}
+
+export class ReadMeDocsUrl extends DocsUrlBase {
+    type: "readme" = "readme";
+
+    async getAllOpenApiUrls(): Promise<string[]> {
+        const origin = this.getBaseUrl();
+        const response = await axios.get(origin + "/openapi");
+
+        if (response.status / 100 !== 2) {
+            return [];
+        }
```

Comment:
what is this doing?

Reply:
Apologies, this was by instinct (handling all non-2XX here). I pushed a commit that should make it more clear: if we don't get a 200, then we log an error and return an empty list, signaling that we were unable to find any openapi urls